Blockzilla/AutocompleteCustomUrlViewController.swift

```diff
@@ -0,0 +1,195 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import SnapKit
+import Telemetry
+
+class AutocompleteCustomUrlViewController: UIViewController {
+    private let emptyStateView = UIView()
+    private let tableView = UITableView(frame: .zero, style: .grouped)
+    private var addDomainCell: UITableViewCell?
+
+    private let customAutocompleteSource: CustomAutocompleteSource
+    private var domains: [String] { return customAutocompleteSource.getSuggestions() }
+
+    init(customAutocompleteSource: CustomAutocompleteSource) {
+        self.customAutocompleteSource = customAutocompleteSource
+        super.init(nibName: nil, bundle: nil)
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: UIConstants.strings.edit, style: .plain, target: self, action: #selector(AutocompleteCustomUrlViewController.toggleEditing))
+
+        view.addSubview(tableView)
+
+        let label = UILabel()
+        label.text = UIConstants.strings.autocompleteEmptyState
+        label.font = UIConstants.fonts.settingsDescriptionText
+        label.textColor = UIConstants.colors.settingsTextLabel
+        label.textAlignment = .center
+        emptyStateView.addSubview(label)
+        tableView.backgroundView = emptyStateView
+        tableView.backgroundView?.isHidden = true
+
+        label.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalToSuperview().offset(50)
+        }
+
+        tableView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        view.backgroundColor = UIConstants.colors.background
+
+        title = UIConstants.strings.autocompleteCustomSectionLabel
+
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.backgroundColor = UIConstants.colors.background
+        tableView.layoutMargins = UIEdgeInsets.zero
+        tableView.separatorColor = UIConstants.colors.settingsSeparator
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadData()
+    }
+
+    @objc private func toggleEditing() {
+        navigationItem.rightBarButtonItem?.title = tableView.isEditing ? UIConstants.strings.edit : UIConstants.strings.done
+        
+        tableView.setEditing(!tableView.isEditing, animated: true)
+        addDomainCell?.animateHidden(tableView.isEditing, duration: 0.2)
+        updateEmptyStateView()
+    }
+
+    @objc fileprivate func updateEmptyStateView() {
+        if tableView.isEditing && domains.isEmpty {
```

Comment:
Should we disabled the 'edit' button when there are no domains in the list?

Reply:
Thats a good question, @lime124?